src/BigT/bigt.java

```diff
@@ -0,0 +1,509 @@
+package BigT;
+
+import java.io.*;
+import java.util.ArrayList;
+
+import bufmgr.*;
+import global.*;
+import btree.*;
+import heap.*;
+import iterator.*;
+import index.*;
+import java.util.*;
+
+public class bigt {
+    private String name;
+    private int type;
+    public ArrayList<Heapfile> heapFiles;
+    public ArrayList<String> heapFileNames;
+    public ArrayList<String> indexFileNames;
+    public ArrayList<BTreeFile> indexFiles;
+    public BTreeFile utilityIndex = null;
+    public Heapfile hf=null;
+    public String indexUtil;
+    private AttrType[] attrType;
+    private FldSpec[] projlist;
+    private CondExpr[] expr;
+    MapIndexScan index_scan;
+    private int insertType;
+    short[] res_str_sizes = new short[]{Map.DEFAULT_ROW_LABEL_ATTRIBUTE_SIZE, Map.DEFAULT_STRING_ATTRIBUTE_SIZE, Map.DEFAULT_STRING_ATTRIBUTE_SIZE};
+
+    public bigt(String name, int type) throws HFException, HFBufMgrException, HFDiskMgrException, IOException,
+            GetFileEntryException, ConstructPageException, AddFileEntryException, btree.IteratorException,
+            btree.UnpinPageException, btree.FreePageException, btree.DeleteFileEntryException, btree.PinPageException,
+            PageUnpinnedException, InvalidFrameNumberException, HashEntryNotFoundException, ReplacerException {
+        String fileName = "";
+        heapFiles = new ArrayList<>(6);
+        indexFiles = new ArrayList<>(6);
+        heapFileNames = new ArrayList<>(6);
+        indexFileNames = new ArrayList<>(6);
+        this.name = name;
+        this.type = type;
+        boolean insert = (type ==1);
```

Comment:
type is an integer be- tween 1 and 5 and the different types will correspond to different clustering and indexing strategies you will use for the bigtable.